requirements-dev.txt

```diff
@@ -17,4 +17,5 @@ click==8.0.4
 cmake
 custatevec-cu12
 pylint
-scipy
+scipy==1.12.0
+quimb==1.7.3
```

Comment:
This is an outdated file (the pin now is only on `scipy`).

At some point, there was [an error](https://github.com/PennyLaneAI/pennylane-lightning/actions/runs/8637854274/job/23681014172) coming from quimb. Eventually, I realized it was due to the wrong version of scipy (also used by quimb). Therefore, if I remove the pin from `scipy`, we get failures from quimb

Reply:
We should probably request this as a pin in quimb, rather than here, but fine for now. Also, if we can pin as `scipy~=1.12.0`  that would still bring in bugfix releases. Let's make a story about tracking this, as I'd rather avoid the pin long-term.

Reply:
We should probably request this as a pin in quimb, rather than here, but fine for now. Also, if we can pin as `scipy~=1.12.0` that would still bring in bugfix releases. Let's make a story about tracking this, as I'd rather avoid the pin long-term.